File: 643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int left=  0;
        double ml = 0;
        double sum = 0;
        for(int i = 0 ; i < k ; i++){
            sum += nums[i];
        }

        ml = sum;
        for(int i = k ; i < nums.length ; i++){
            sum += nums[i] - nums[i - k];
            ml = Math.max(sum , ml);
        }
        return ml / k;
    }
}

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution(object):
    def findMaxAverage(self, nums, k):
        left = 0
        maxs = float('-inf')
        wind = 0
        for right in range(len(nums)):
            wind += nums[right]
            if right - left + 1 == k:
                maxs = max(maxs , wind)
                wind -= nums[left]
                left += 1
        return maxs * 1.0/ k
        